### PR TITLE
SS4 Compatibility Changes

### PR DESCRIPTION
**Pull request by ssmarco**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,53 +1,33 @@
...
-php:
-  - 5.6
-  - 7.0
+sudo: false
 
 env:
   global:
-    - DB=MYSQL CORE_RELEASE=3
+    - COMPOSER_ROOT_VERSION=1.0.x-dev
+    - CORE_RELEASE=master
```
Comment (dhensby) [inline .travis.yml]: core release should be 4

Comment (marczhermo) [inline .travis.yml]: Thanks for looking into it. Will update the soonest 😀

File: tests/IFramePageTest.php [hunk 2]
```diff
@@ -101,38 +104,38 @@ public function testForceProtocol()
...
         $this->assertNull($response);
 
-        Config::inst()->update('Director', 'alternate_protocol', 'http');
-        Config::inst()->update('Director', 'alternate_base_url', 'http://host.com');
+        Config::inst()->update('SilverStripe\Control\Director', 'alternate_protocol', 'http');
+        Config::inst()->update('SilverStripe\Control\Director', 'alternate_base_url', 'http://host.com');
         $page->ForceProtocol = 'http://';
-        $controller = new IFramePage_Controller($page);
+        $controller = new IFramePageController($page);
         $response = $controller->init();
```
Comment (robbieaverill) [inline tests/IFramePageTest.php]: `Controller::init` is now (in 4.x) protected, you should use `doInit` instead

Comment (ssmarco) [inline tests/IFramePageTest.php]: Thanks for looking into this. Have updated my pull request.

File: composer.json [hunk 3]
```diff
@@ -17,6 +17,12 @@
 	"require-dev": {
 		"phpunit/PHPUnit": "^5.7"
 	},
+	"autoload": {
+        "psr-4": {
+            "SilverStripe\\IFrame\\": "code/",
+            "SilverStripe\\IFrame\\Tests\\": "tests/"
```
Comment (robbieaverill) [inline composer.json]: The test namespaces should be under "autoload-dev"

File: code/IFramePageController.php [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+namespace SilverStripe\IFrame;
+
+use SilverStripe\CMS\Controllers\ContentController;
+use SilverStripe\Control\Director;
+use SilverStripe\View\Requirements;
+
+class IFramePageController extends ContentController
+{
+    public function init()
```
Comment (robbieaverill) [inline code/IFramePageController.php]: Should be protected to match `ContentController::init`

Comment (robbieaverill) [inline code/IFramePageController.php]: Should be protected to match `ContentController::init`

Comment (ssmarco) [inline code/IFramePageController.php]: Sure thing, thanks!

File: composer.json (hunk 3 above)
Comment (tractorcow) [inline composer.json]: not for modules; Modules are rarely installed in root, and autoload-dev is root only. This means if you add iframe to your project, and ran test suite, test classes won't be psr-4 autoloaded.

File: composer.json [hunk 5]
```diff
@@ -11,12 +11,27 @@
...
 	},
+	"require-dev": {
+		"phpunit/PHPUnit": "^5.7"
+	},
+	"autoload": {
+        "psr-4": {
+            "SilverStripe\\IFrame\\": "code/"
+        }
+    },
+    "autoload-dev": {
```
Comment (tractorcow) [inline composer.json]: Sorry to contradict @robbieaverill , but this does need to be in non-dev autoload.

Comment (robbieaverill) [inline composer.json]: My mistake - good to know - sorry @ssmarco

File: lang/en.yml [hunk 6]
```diff
@@ -3,11 +3,9 @@ en:
...
     PLURALNAME: 'Base Pages'
     SINGULARNAME: 'I Frame Page'
-    VALIDATION:
-      BANNEDURLSCHEME: 'This URL scheme is not allowed.'
-  IframePage:
+    VALIDATION_BANNEDURLSCHEME: 'This URL scheme is not allowed.'
     ExternalNote: 'Please note the following section of content is possibly being delivered from an external source (IFRAME in HTML terms), and may present unusual experiences for screen readers.'
     Loading: 'Loading content...'
-  SiteTree:
+  SilverStripe\CMS\Model\SiteTree:
```
Comment (tractorcow) [inline lang/en.yml]: Should be removed. This belongs to CMS module.

File: composer.json (hunk 5 above)
Comment (ssmarco) [inline composer.json]: No worries, will change this as soonest. :)

File: lang/en.yml (hunk 6 above)
Comment (ssmarco) [inline lang/en.yml]: How do I do this?

**Comment by robbieaverill** [comment]
@ssmarco do you want to rebase this PR and tidy up some of the duplicate/unnecessary commit messages and we'll get this merged?

**Pull request merged**